### ISSUE TITLE
Warning for unseen categories {nan}

### ISSUE THREAD

**Issue by ianlokh** [sdv-dev/RDT#596]
### Environment Details

Please indicate the following details about the environment in which you found the bug:

* RDT version: 1.2.0
* Python version: 3.8.13
* Operating System: OSX 12.6.1

### Error Description

When running categorical transformation, the following warning was encountered -

/Users/stb/opt/anaconda3/envs/torch1130_py38/lib/python3.8/site-packages/rdt/transformers/categorical.py:379: UserWarning: The data contains 1 new categories that were not seen in the original data (examples: {nan}). Creating a vector of all 0s. If you want to model new categories, please fit the transformer again with the new data.

This occurred when I was executing a transformation for a categorical column with NaNs

### Steps to reproduce

1. Load a dataset with a column of categorical values of 1.0, 2.0, 3.0 and NaN
2. Create a OneHotEncoder instance (as transformer) and fit the dataset.
3. Save the instance with transformer.save
4. Create another dataset with the same columns as (1)
5. Load the transformer instance
6. Run transformer.transform on new dataset.

```
 warnings.warn(
/Users/stb/opt/anaconda3/envs/torch1130_py38/lib/python3.8/site-packages/rdt/transformers/categorical.py:379: UserWarning: The data contains 1 new categories that were not seen in the original data (examples: {nan}). Creating a vector of all 0s. If you want to model new categories, please fit the transformer again with the new data.

```

**Comment by npatki** [comment]
Hi @ianlokh, thanks for filing! The error may be indicating that you are using an unfit transformer. There are a few potential areas where this may be happening --

1. **Saving/Loading a transformer**: This functionality isn't yet supported through any RDT API -- eg. there isn't a `.save` function available for any transformer class or HyperTransformer.
2. **Transformer usage**: Although it's possible to use transformers in a standalone way, the recommended usage is to create and use HyperTransformer instead. See code snippet below.

```python
import numpy as np
import pandas as pd
from rdt import HyperTransformer
from rdt.transformers.categorical import OneHotEncoder

data = pd.DataFrame(data={
    'column_name': [1.0, 2.0, np.nan, 2.0, 3.0, np.nan, 3.0] 
})

ht = HyperTransformer()
ht.set_config({
    'sdtypes': { 'column_name': 'categorical' },
    'transformers': { 'column_name': OneHotEncoder() }
})
ht.fit(data)
ht.transform(data)
```

Could you provide a code snippet that shows how you went through the replication steps?

**Comment by npatki** [comment]
Hi @candalfigomoro and @ianlokh thanks for the details and input. I was able to replicate issue and trace it back to some points of the code. I've filed bug  #616 with the details. Let's use that bug to track the fix .

The good news is that, based on my explorations, **the warning is unnecessary**.  It seems that the transformer is working -- ie it is transforming and reverse transforming the `nan` values just fine. Feel free to use #616 to continue the conversation.
